src/Minimap.ts

```diff
@@ -0,0 +1,288 @@
+/**
+ * This is an extension adds support for adding a minimap to one of the map's control containers.
+ */
+
+import { Map } from "./Map";
+import { DOMcreate, DOMremove } from "./tools";
+
+import type {
+  ControlPosition,
+  FillLayerSpecification,
+  GeoJSONSource,
+  IControl,
+  LineLayerSpecification,
+  StyleOptions,
+  StyleSpecification,
+  StyleSwapOptions,
+} from "maplibre-gl";
+import type { MapOptions } from "./Map";
+import type { MapStyleVariant, ReferenceMapStyle } from "@maptiler/client";
+
+export interface ParentRect {
+  lineLayout: LineLayerSpecification["layout"];
+  linePaint: LineLayerSpecification["paint"];
+  fillPaint: FillLayerSpecification["paint"];
+}
+
+export interface MinimapOptionsInput {
+  /**
+   * Style of the map. Can be:
+   * - a full style URL (possibly with API key)
+   * - a shorthand with only the MapTIler style name (eg. `"streets-v2"`)
+   * - a longer form with the prefix `"maptiler://"` (eg. `"maptiler://streets-v2"`)
+   */
+  style?: ReferenceMapStyle | MapStyleVariant | StyleSpecification | string;
+
+  /**
+   * Set the zoom difference between the parent and the minimap
+   * If the parent is zoomed to 10 and the minimap is zoomed to 8, the zoomAdjust should be 2
+   * Default: -4
+   */
+  zoomAdjust?: number;
```

Comment:
I like that when this value is a positive number, the minimap is actually a magnifier!